openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -289,6 +289,13 @@ def has_ended(self):
         """
         return course_metadata_utils.has_course_ended(self.end)
 
+    def starts_within(self, days):
+        """
+        Returns whether the course starts with-in given days.
```

Comment:
I think doc string of this method can be improved too. Something like "Returns a boolean stating if course starts with-in given number of days"